cli/tests/environment-pull.bats

```diff
@@ -181,3 +215,47 @@ function update_dummy_env() {
   run "$FLOX_BIN" pull --remote floxtest/default
   assert_success
 }
+
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=pull:unsupported
+# pulling an environment without packages for the current platform
+#should fail with an error
+@test "l?: pull environment without packages for the current platform fails" {
+  make_dummy_env "owner" "name"
+  update_dummy_env "owner" "name"
+  make_incompatible "owner" "name"
+
+  run "$FLOX_BIN" pull --remote owner/name
+  assert_failure
+  assert_output --partial "This environment is not yet compatible with your system ($NIX_SYSTEM)"
+
+  run "$FLOX_BIN" pull --remote owner/name --add-system
+  assert_success
```

Comment:
nonblocking suggestion: update the test name or make this a separate test

Reply:
split